### PR TITLE
native/mtd: implement .write_page()

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

`native` didn't implement the new pagewise API yet, so unaligned writes / writes across page boundaries will fail.

### Issues/PRs references

https://github.com/RIOT-OS/RIOT/pull/15362#issuecomment-721378089

### CODE REVIEW

**Comment by jue89** [comment]
Does it really make sense to implement `.write` and `.write_page` separately?

The `mtd` driver would translate between both worlds if I understand its implementation correctly. So to me this is more or less redundant code - we don't loose functionality if we'd get ride of `.write`.

**Comment by jue89** [comment]
But all in all this PR fixes  https://github.com/RIOT-OS/RIOT/pull/15362#issuecomment-721378089 :)

```
git checkout benpicco/tests/mtd_raw
git checkout -b mtd_raw_test
git merge benpicco/native/mtd_pagewise
make -C tests/mtd_raw BOARD=native all term
(...)
/home/jue/Projects/RIOT/tests/mtd_raw/bin/native/tests_mtd_raw.elf  
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2021.01-devel-547-g4caa95-mtd_raw_test)
Manual MTD test
init MTD_0… OK (8192 kiB)
> test 0
test 0
[START]
[SUCCESS]
```

**Comment by benpicco** [comment]
Heh, looks like the unit test expects mtd_write() to fail on unaligned writes.
I'll move the 2nd commit to the other PR.

File: cpu/native/mtd/mtd_native.c [hunk 1]
```diff
@@ -103,6 +105,41 @@ static int _write(mtd_dev_t *dev, const void *buff, uint32_t addr, uint32_t size
...
+static int _write_page(mtd_dev_t *dev, const void *buff, uint32_t page, uint32_t offset,
+                       uint32_t size)
+{
+    mtd_native_dev_t *_dev = (mtd_native_dev_t*) dev;
+    uint32_t addr = page * dev->page_size + offset;
+
+    DEBUG("mtd_native: write from page %" PRIx32 ", offset 0x%" PRIx32 " count %" PRIu32 "\n",
+          page, offset, size);
+
+    if (page > dev->sector_count * dev->pages_per_sector) {
```
Comment (bergzand) [inline cpu/native/mtd/mtd_native.c:117]: Why not just call `_write()` here?

Comment (benpicco) [reply cpu/native/mtd/mtd_native.c:117]: I'd like to get rid of `_write()` eventually

**Review by bergzand — APPROVED** [review]
Ack! Tested with `tests/pkg_littlefs`

**Pull request merged**